### PR TITLE
add memoryutil metric

### PR DESCRIPTION
**Pull request by boyang9527**
[#144957125]

### CODE REVIEW

**Comment by rohitsharma04** [comment]
@boyang9527 travis build is failing there's data race in one of the integration tests.

File: api/lib/validation/schemaValidator.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ var getDaysInMonthInISOFormat = function() {
 };
 
 var getMetricTypes = function() {
-  var metricTypeEnum = ['memoryused'];
+  var metricTypeEnum = ['memoryused', 'memoryutil', 'responsetime', 'throughput']
```
Comment (qibobo) [inline api/lib/validation/schemaValidator.js]: Should we read the metric types from config file?

File: src/autoscaler/db/sqldb/appmetric_sqldb_test.go [hunk 2]
```diff
@@ -76,8 +76,8 @@ var _ = Describe("AppMetricSQLDB", func() {
 			BeforeEach(func() {
 				appMetric := &models.AppMetric{
 					AppId:      "test-app-id",
-					MetricType: models.MetricNameMemory,
-					Unit:       models.UnitMegaBytes,
+					MetricType: "test-metric-name",
```
Comment (qibobo) [inline src/autoscaler/db/sqldb/appmetric_sqldb_test.go]: The two strings "test-metric-name" and "test-metric-unit" appear many times in these test cases, is it better to define them as constant?

File: api/lib/validation/schemaValidator.js (hunk 1 above)
Comment (boyang9527) [inline api/lib/validation/schemaValidator.js]: These are the metrics we can support with the events we get from loggregator. Once we support custom metrics, we will not have the metric type check anymore... So I would keep this way for now.

File: src/autoscaler/db/sqldb/appmetric_sqldb_test.go (hunk 2 above)
Comment (boyang9527) [inline src/autoscaler/db/sqldb/appmetric_sqldb_test.go]: done in another commit to refactor the test cases

**Comment by qibobo** [comment]
LGTM

File: src/autoscaler/eventgenerator/aggregator/policyPoller_test.go [hunk 3]
```diff
@@ -79,13 +79,13 @@ var _ = Describe("PolicyPoller", func() {
...
-					Expect(policyMap[testAppId1]).To(Equal(&models.AppPolicy{
-						AppId: testAppId1,
+					Expect(policyMap[testAppId]).To(Equal(&models.AppPolicy{
+						AppId: testAppId,
 						ScalingPolicy: &models.ScalingPolicy{
 							InstanceMax: 5,
 							InstanceMin: 1,
 							ScalingRules: []*models.ScalingRule{{
-								MetricType:            models.MetricNameMemory,
+								MetricType:            "test-metric-name",
```
Comment (rohitsharma04) [inline src/autoscaler/eventgenerator/aggregator/policyPoller_test.go]: use constant.

File: src/autoscaler/scalingengine/server/scaling_handler_test.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ var _ = Describe("ScalingHandler", func() {
 				scalingEngine.ScaleReturns(3, nil)
 
 				trigger = &models.Trigger{
-					MetricType: models.MetricNameMemory,
+					MetricType: "test-metric-name",
```
Comment (rohitsharma04) [inline src/autoscaler/scalingengine/server/scaling_handler_test.go]: use contant.

Comment (boyang9527) [inline src/autoscaler/scalingengine/server/scaling_handler_test.go]: there are two places using "test-metric-name", changed to a variable

**Comment by boyang9527** [comment]
@rohitsharma04 did a force push with changes

File: api/lib/validation/schemaValidator.js (hunk 1 above)
Comment (boyang9527) [inline api/lib/validation/schemaValidator.js]: fixed

File: src/autoscaler/eventgenerator/aggregator/policyPoller_test.go (hunk 3 above)
Comment (boyang9527) [inline src/autoscaler/eventgenerator/aggregator/policyPoller_test.go]: modes.??? constant is not used to test generic metric name.  There is only one place using string "test-metric-name",  it is not worth defining it as a constant or variable.

**Comment by boyang9527** [comment]
@rohitsharma04  given the acceptance test is green, can you review the changes again?  this PR has blocked several PRs in release project for UAT of  scaling based on memoryutil/throughput/response time.
